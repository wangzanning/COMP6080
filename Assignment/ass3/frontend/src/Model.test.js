import { shallow } from 'enzyme';
import React from 'react';
import Modal from './screens/components/modal';

describe('Modal', () => {
  let modalStart, modalStop, props, propsStop, propsNotShow, modalNotShow;

  beforeAll(() => {
    props = {
      show: true,
      close: jest.fn(),
      content: `Session ID is 11111`,
      operation: 'Copy Link',
      copy: jest.fn(),
      sessionId: 11111
    };
    propsStop = {
      show: true,
      close: jest.fn(),
      content: 'Would you like to view the results?',
      operation: 'Yes',
      sessionId: 11111
    }
    propsNotShow = {
      show: false
    }
    modalStart = shallow(<Modal {...props} />);
    modalStop = shallow(<Modal {...propsStop} />);
    modalNotShow = shallow(<Modal {...propsNotShow} />);
  });

  // Testing snapshot
  it('is matched with snapshot', () => {
    expect(modalStart).toMatchSnapshot();
    expect(modalStop).toMatchSnapshot();
    expect(modalNotShow).toMatchSnapshot();
  });

  // Testing props
  it('can display correct content when start game', () => {
    expect(modalStart.find('p').text()).toEqual('Session ID is 11111');
  });
  it('can display correct content when stop game', () => {
    expect(modalStop.find('p').text()).toEqual('Would you like to view the results?');
  });



  // Testing buttons
  it('should have three buttons when started', () => {
    expect(modalStart.find('button').length).toBe(3);
    expect(modalStart.find('button').at(0).text()).toEqual('Join Game');
    expect(modalStart.find('button').at(1).text()).toEqual('Copy Link');
    expect(modalStart.find('button').at(2).text()).toEqual('Close');
  });

  it('should have two buttons when closed', () => {
    expect(modalStop.find('button').length).toBe(2);
    expect(modalStop.find('button').at(0).text()).toEqual('Yes');
    expect(modalStop.find('button').at(1).text()).toEqual('Close');
  });

  it('close button can trigger onClick function from GameCard', () => {
    modalStart.find('button').at(2).simulate('click');
    expect(props.close).toHaveBeenCalled();
    modalStop.find('button').at(1).simulate('click');
    expect(propsStop.close).toHaveBeenCalled();
  });

  it('copy button can trigger function', () => {
    modalStart.find('button').at(1).simulate('click');
    expect(props.copy).toHaveBeenCalled();
  })

});