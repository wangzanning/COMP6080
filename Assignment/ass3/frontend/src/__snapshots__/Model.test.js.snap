// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is matched with snapshot 1`] = `ShallowWrapper {}`;

exports[`Modal is matched with snapshot 2`] = `ShallowWrapper {}`;

exports[`Modal is matched with snapshot 3`] = `ShallowWrapper {}`;
